lib/keyfile/keyfile.c

```diff
@@ -372,27 +372,31 @@ keyfile_read(const char * filename, uint64_t * machinenum, int keys)
 	struct stat sb;
 	uint8_t * keybuf;
 	FILE * f;
+	size_t keyfilelen;
 
 	/* Stat the file. */
 	if (stat(filename, &sb)) {
 		warnp("stat(%s)", filename);
 		goto err0;
 	}
 
-	/* Allocate memory. */
-	if ((sb.st_size == 0) || (sb.st_size > 1000000)) {
+	/* Validate keyfile size. */
+	keyfilelen = (size_t)(sb.st_size);
```

Comment:
Parentheses are good.  The order of operations between `.` and `(type)` operators isn't something most people have memorized.